### DISCUSSION TITLE
Using an internal voltage reference

### DISCUSSION THREAD

**Discussion by ghost** [stm32duino/Arduino_Core_STM32#1578] [Q&A] [answered]
Hello everyone! I am using STM32L152CBT6 microcontroller. Tell me if this microcontroller has an internal reference voltage, and how to use it for the ADC?
Thanks!

**Comment by ABOSTM** [comment, marked as answer]
Hi @HomeMade-Channel,
To know what are the features supported by your MCU, you can have a look at Reference Manual or Datasheet.
To know how to use internal reference with ADC, you can have a look at wiki which point to an example `Internal_channels`:
https://github.com/stm32duino/wiki/wiki/API#example-2
